spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -1455,6 +1457,31 @@ public void testUpdateToWapBranchWithTableBranchIdentifier() {
                         branch)));
   }
 
+  @Test
+  public void subExpressionEliminationInCodegen() {
+    createAndInitTable("id INT, c1 INT, c2 INT");
+
+    sql("INSERT INTO TABLE %s VALUES (1, 11, 111), (2, 22, 222)", tableName);
+    createBranchIfNeeded();
+
+    // disable AQE to see the final plan with codegen in EXPLAIN
```

Comment:
can we do something like : 
```
   def getCodeAndCommentForUpdateRowExec(df: DataFrame): CodeAndComment = {
    val plan = df.queryExecution.executedPlan
    plan.execute()
    val updateRowExec = findTopLevelUpdateRowExec(plan)
    getCodeAndComment(updateRowExec.head)
  }

  def findTopLevelUpdateRowExec(plan: SparkPlan): Seq[UpdateRowExec] = {
    filterByType[UpdateRowExec](plan)
  }

  private def getCodeAndComment(plan: SparkPlan): CodeAndComment = {
    val codeGenSubTree = WholeStageCodegenExec(plan)(1)
    val codeAndComment = codeGenSubTree.doCodeGen()._2
    try {
      CodeGenerator.compile(codeAndComment)
    } catch {
      case e: Exception =>
        val msg =
          s"""
             |failed to compile:
             |Subtree:
             |$codeGenSubTree
             |Generated code:
             |${CodeFormatter.format(codeAndComment)}
            """.stripMargin
        fail(msg, e)
    }
    codeAndComment
  }
```

This way we can test both AQE without AQE plans, thoughts @aokolnychyi  ?

Reply:
Let me play around.